Explorer/Assets/Scripts/CrdtEcsBridge/Engine/SimpleFetchApiImplementation.cs

```diff
@@ -0,0 +1,121 @@
+            var webRequestHeaders = new WebRequestHeadersInfo();
+            headersDictionary.Clear();
+
+            if (headers is IScriptObject scriptObject)
+            {
+                IEnumerable<string> propertyNames = scriptObject.PropertyNames;
+
+                foreach (string name in propertyNames)
+                {
+                    var property = scriptObject.GetProperty(name).ToString();
+                    webRequestHeaders.Add(name, property);
+                    headersDictionary.Add(name, property);
+                }
+            }
+
+            ITypedWebRequest request;
+
+            switch (parsedRequestMethod)
+            {
+                case RequestMethod.GET:
+                    request = await webController.GetAsync(commonArguments, ct, ReportCategory.SCENE_FETCH_REQUEST, webRequestHeaders);
+                    GenerateResponse(request);
+                    break;
+                case RequestMethod.POST:
+                    headersDictionary.TryGetValue("content-type", out string postContentType);
+                    var postArguments = GenericPostArguments.Create(body, postContentType ?? "");
+                    request = await webController.PostAsync(commonArguments, postArguments, ct, ReportCategory.SCENE_FETCH_REQUEST, webRequestHeaders);
+                    GenerateResponse(request);
+                    break;
+                case RequestMethod.PUT:
+                    headersDictionary.TryGetValue("content-type", out string putContentType);
+                    var putArguments = GenericPutArguments.Create(body, putContentType ?? "");
+                    request = await webController.PutAsync(commonArguments, putArguments, ct, ReportCategory.SCENE_FETCH_REQUEST, webRequestHeaders);
+                    GenerateResponse(request);
+                    break;
+                case RequestMethod.PATCH:
+                    headersDictionary.TryGetValue("content-type", out string patchContentType);
+                    var patchArguments = GenericPatchArguments.Create(body, patchContentType ?? "");
+                    request = await webController.PatchAsync(commonArguments, patchArguments, ct, ReportCategory.SCENE_FETCH_REQUEST, webRequestHeaders);
+                    GenerateResponse(request);
+                    break;
+                case RequestMethod.INVALID:
+                default: throw new ArgumentOutOfRangeException();
+            }
+
+            return new ISimpleFetchApi.FetchResponse();
+        }
+
+        private ISimpleFetchApi.FetchResponse GenerateResponse(ITypedWebRequest request)
+        {
+            UnityWebRequest unityWebRequest = request.UnityWebRequest;
+            string responseData = unityWebRequest.downloadHandler?.text ?? string.Empty;
+
+            var headers = new List<ISimpleFetchApi.Header>();
+
+            foreach (string headerKey in unityWebRequest.GetResponseHeaders().Keys)
```

Comment:
It's very allocation unfriendly and you don't use the return value - just comment it out with some explanation that the return value is not used anyway rather than producing allocations for nothing.

Reply:
Actually headers is part of the FetchResponse ->
`            return new ISimpleFetchApi.FetchResponse(headers, ok, redirected, status, statusText, url, responseData);`
Thou I agree its allocation unfriendly... I will see what I can do :)